gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java

```diff
@@ -0,0 +1,215 @@
+        ByteArrayOutputStream aos = new ByteArrayOutputStream();
+        ObjectOutputStream oos = new ObjectOutputStream(aos)) {
+
+      oos.writeObject(spec);
+      statement.setString(1, spec.getUri().toString());
+      statement.setBlob(2, new ByteArrayInputStream(aos.toByteArray()));
+      statement.executeUpdate();
+
+      connection.commit();
+    } catch (SQLException e) {
+      throw new IOException(e);
+    }
+  }
+
+  @Override
+  public boolean deleteSpec(Spec spec) throws IOException {
+    return deleteSpec(spec.getUri());
+  }
+
+  @Override
+  public boolean deleteSpec(URI specUri) throws IOException {
+    try (Connection connection = this.dataSource.getConnection();
+        PreparedStatement statement = connection.prepareStatement(String.format(DELETE_STATEMENT, this.tableName))) {
+      statement.setString(1, specUri.toString());
+      int result = statement.executeUpdate();
+      connection.commit();
+      return result != 0;
+    } catch (SQLException e) {
+      throw new IOException(e);
+    }
+  }
+
+  @Override
+  public boolean deleteSpec(URI specUri, String version) throws IOException {
+    return deleteSpec(specUri);
+  }
+
+  @Override
+  public Spec updateSpec(Spec spec) throws IOException, SpecNotFoundException {
+    addSpec(spec);
+    return spec;
+  }
+
+  @Override
+  public Spec getSpec(URI specUri) throws IOException, SpecNotFoundException {
+    try (Connection connection = this.dataSource.getConnection();
+        PreparedStatement statement = connection.prepareStatement(String.format(GET_STATEMENT, this.tableName))) {
+
+      statement.setString(1, specUri.toString());
+      ResultSet rs = statement.executeQuery();
+      if (!rs.next()) {
+        throw new SpecNotFoundException(specUri);
+      }
+
+      Blob blob = rs.getBlob(1);
+
+      try (ObjectInputStream ois = new ObjectInputStream(blob.getBinaryStream())) {
+        return (Spec) ois.readObject();
+      }
+
+    } catch (SQLException | ClassNotFoundException e) {
+      throw new IOException(e);
+    }
+  }
+
+  @Override
+  public Spec getSpec(URI specUri, String version) throws IOException, SpecNotFoundException {
+    return getSpec(specUri);
+  }
+
+  @Override
+  public Collection<Spec> getAllVersionsOfSpec(URI specUri) throws IOException, SpecNotFoundException {
+    return Lists.newArrayList(getSpec(specUri));
+  }
+
+  @Override
+  public Collection<Spec> getSpecs() throws IOException {
+    try (Connection connection = this.dataSource.getConnection();
+        PreparedStatement statement = connection.prepareStatement(String.format(GET_ALL_STATEMENT, this.tableName))) {
+
+      List<Spec> specs = new ArrayList<>();
+      ResultSet rs = statement.executeQuery();
+
+      while (rs.next()) {
+        Blob blob = rs.getBlob(2);
+        try (ObjectInputStream ois = new ObjectInputStream(blob.getBinaryStream())) {
+          specs.add((Spec) ois.readObject());
```

Comment:
Should we catch any exceptions while de-serializing an individual spec? We can log an error and return the successfully de-serialized specs?

Reply:
I wasn't sure whether to do this or not, but you're right, it's probably better not to fail everything if there's just one corrupt spec.